### PR TITLE
Pass proper COM interfaces to the drag functions

### LINKED ISSUE #6952: Winforms runtime has stopped working when calling control's dragdrop event
### .NET version

.Net 7.0.100-preview.4.22201.11

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, it works well with previous .Net 7.0 SDK build: 7.0.100-**preview.3**.22153.22, 7.0.100-**preview.4**.22173.7

### Issue description

Winforms runtime has stopped working when calling control's dragdrop event, see following gif:

7 0_Issue

### Steps to reproduce

1. Use attached sample application:
[WinFormsApp30.zip](https://github.com/dotnet/winforms/files/8402059/WinFormsApp30.zip)
2. Build and run application
3. Drag and drop item from the left listBox to the right listBox

### PR DESCRIPTION
**Pull request by kant2002**
Fixes #6952

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6981)

### CODE REVIEW

**Comment by RussKie** [comment]
Let's add a regression test to this using the repro provided in https://github.com/dotnet/winforms/issues/6952.
The test should go into https://github.com/dotnet/winforms/tree/main/src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests, and we can programmatically perform drag and drop with the help of `InputSimulator`.

**Review by JeremyKuhne — DISMISSED** [review]
Outside of adding a regression test as @RussKie suggested, this looks good. We should also consider making a helper method to get a specific interface from WinFormsComWrappers.

**Comment by kant2002** [comment]
@RussKie all delays seems to be important. If I drag too fast, test become unreliable.

**Comment by kant2002** [comment]
@JeremyKuhne can you give this another pass of review?

**Review by JeremyKuhne — DISMISSED** [review]
Looks good

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DragDropTests.cs [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+        }
+
+        private void ListDragSource_QueryContinueDrag(object? sender, QueryContinueDragEventArgs e)
+        {
+            testOutputHelper.WriteLine($"Query for drag continuation.");
+
+            // Cancel the drag if the mouse moves off the form.
+            if (sender is ListBox lb)
+            {
+                Form f = lb.FindForm();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DragDropTests.cs:329]: nit: `f`->`form`

Comment (kant2002) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DragDropTests.cs]: Thanks! Fixed.

**Review by Tanya-Solyanik — APPROVED** [review]

**Comment by kant2002** [comment]
@dreddy-work can I ask you to merge this?

**Pull request merged**